1a.cpp

```diff
@@ -0,0 +1,55 @@
+#include <iostream>
+#include <vector>
+
+int main()
+{
+	std::cout << "Введите размер вектора\n";
+	int vecsize;
+	std::cin >> vecsize;
+	if (std::cin.fail())
+	{
+		std::cout << "Введено нецелочисленное значение.\n";
+		return 0;
+	}
+	else if (vecsize <= 0)
+	{
+		do
+		{		
+			std::cout << "Размер вектора должен быть положителен. Введите значение заново.\n";
+			std::cin >> vecsize;
+		}
+		while (vecsize <= 0);
+	}
+	std::vector<int> v1;
+	std::cout << "Коллекция чисел: \n";
+	for(unsigned int i = 0; i < vecsize; i++)
+	{
+		int j = i;
+		int value = (vecsize - i) * 2;
+		v1.push_back(value);
+		std::cout << ++j << "е число " << v1.back() << "\n";
+	}
+	std::cout << "Размер вектора: " << v1.size() << "\n";
+
+
+	for (int j = 0; j < vecsize; j++)
+	{
+		int posit = j;
```

Comment:
Why is this variable declared? Why not just use `j`?

Reply:
Because 'posit' changes during the cycle, while j does not.
![src1a](https://user-images.githubusercontent.com/25636404/30022972-5076a322-9176-11e7-8830-a72f4d352fae.png)